### PR TITLE
include nvm and travis ci config, build with node 8/npm 5

### PR DESCRIPTION
**Pull request by bcruddy**
In reference to #24.

This includes:
- `.nvmrc` to point at node 8
- `package-lock.json` as a part of npm 5
- travis config

I'd be happy to swap out node versions or another CI. I also obviously cannot set up the hooks for travis to run in this repo so if we do go the travis route that will need to be added.

### CODE REVIEW

**Comment by stasm** [comment]
@bcruddy Nice, I love this. Especially the CI part :) Do I need to enable Travis before and after this is merged?

**Comment by bcruddy** [comment]
Yeah if you enable the travis hook we can make sure everything runs as expected (it should run if I force push to the branch) before it gets merged

**Comment by stasm** [comment]
Alright, I've enabled the Travis hook for this repo. Do you want to try it out?

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+v8
```
Comment (stasm) [inline .nvmrc:1]: Is this file specifically for Travis or for using with `nvm use`? Also, should we also add this version to `package.json`?

**Comment by stasm** [comment]
I was able to test the integration by manually triggering a build within the Travis UI: https://travis-ci.org/stasm/innerself/builds/278360298

File: .nvmrc (hunk 1 above)
Comment (bcruddy) [reply .nvmrc:1]: No it's not but Travis does use nvm to set the node version. I tend to work with different node versions across different projects and it's always nice to guarantee I'm using the correct version or switch to it with a simple terminal command. The engine version in package.json should be the minimum version supported.

**Comment by bcruddy** [comment]
@stasm I'm not sure if its just not showing up until someone pushes on an open PR but there's a repository settings to force CI to pass before merging (it can be overridden by admins)

**Pull request merged**